tests/unit/test_charm.py

```diff
@@ -219,16 +248,13 @@ def test_given_statefulset_is_not_patched_when_config_changed_then_statefulset_i
             )
         )
         self.mock_lightkube_client_get.return_value = test_statefulset
-        self.mock_check_output.return_value = b"1.1.1.1"
-        self.harness.add_storage("config", attach=True)
-        self.set_f1_relation_data()
 
         self.harness.update_config(key_values={})
         self.harness.evaluate_status()
 
-        self.mock_lightkube_client_replace.assert_called_once_with(obj=PRIVILEGED_STATEFULSET)
+        assert len(self.mock_lightkube_client_replace.mock_calls) == 2
```

Comment:
I am not a fan of using hard-coded values to test the number of times that something happens, whenever we increase the complexity of the charm .i.e. any other event happens that goes through the `_configure` method (like a relation is joined) we would need to double this value and know that it represents 1 for mounting the volume and 1 for setting the privileged context. I think I would have mocked the classes on `oai_ran_du_k8s.py`

Reply:
I know and I'm not a fan of that either. The only reason why I put it here is that there's something I don't understand in a way mock calls are handled in the tests and this test will be refactored anyway in the next PR (implementing simulation mode). 
But since we're on it anyway, I'll be happy to discuss the problem in more details. Pair programming?

Reply:
Just merged Guillaume's PR which changes the approach to testing.

Reply:
https://docs.python.org/3/library/unittest.mock-examples.html#coping-with-mutable-arguments is helpful if we really need to check call args for mocks.